tesseract_environment/src/utils.cpp

```diff
@@ -388,17 +373,25 @@ bool checkTrajectory(std::vector<tesseract_collision::ContactResultMap>& contact
 
     for (long iStep = start_idx; iStep < end_idx; ++iStep)
     {
+      tesseract_collision::ContactTrajectoryStepResults step_contacts(static_cast<int>(iStep + 1), traj.row(iStep));
+      tesseract_collision::ContactTrajectorySubstepResults substep_contacts(1, traj.row(iStep));
```

Comment:
Same here only resize with the if statement when debug is enabled.

Reply:
This one I am less certain of it's performance, it feels like it's essentially equivalent, but I'm not too familiar with how memory allocation works
```
ContactTrajectoryStepResults::ContactTrajectoryStepResults(const int& step_number, const Eigen::VectorXd& state)
  : step(step_number), state0(state), state1(state), total_substeps(1)
{
  substeps.resize(static_cast<std::size_t>(1));
}
```
But the substep_contact object is the same as above.